### DISCUSSION TITLE
Renovate AMIs regex not working

### DISCUSSION THREAD

**Discussion by asierga95** [renovatebot/renovate#16320] [Request Help] [answered]
Hi,

I am trying to use renovate to keep AMIs updated. I have a file called: "noxfile.py" with the following text on it:
```
# amiFilter=[{"Name":"owner-id","Values":["762873164855"]},{"Name":"name","Values":["ops-base-master-*"]}]
# currentImageName=unknown
base_image_id = "ami-04a726a285fbf6c8d"
```

This is the regex we are using in renovate.json:
```
"regexManagers": [
    {
      "fileMatch": [".*"],
      "matchStrings":
        [
          ".*amiFilter=(?<packageName>.*?)\n(.*currentImageName=(?<currentDigest>.*?)\n)?(.*\n)?.*?(?<depName>[a-zA-Z0-9-_:]*)[ ]*?[:|=][ ]*?[\"|']?(?<currentValue>ami-[a-z0-9]{17})[\"|']?.*"
        ],
      "datasourceTemplate": "aws-machine-image",
      "versioningTemplate": "aws-machine-image"
    }
  ]
```

We have other PullRequest coming from renovate for container images and other updates, but we don't receive the PR for ami update.
When we check the logs of renovate, looks like aws-machine-image is not finding nothing or doing nothing:
The only log we have is printing the regex:

![image](https://user-images.githubusercontent.com/14941467/176398495-0ad703a3-2cc3-474d-8a72-111bbb4f7bc1.png)

Also this regex has been tested in: [regex](https://regex101.com/) and it works fine:
![image](https://user-images.githubusercontent.com/14941467/176398669-59877e56-42cd-49d1-9dd0-d553dbcb3f9f.png)

Just wondering if anyone knows what could be happening there.
Renovate version: 32.83.0

Thanks in advance!!

**Comment by PhilipAbed** [comment]
if you have multiple lines,
maybe try

    "matchStringsStrategy": "combination",
      "matchStrings": [
        "first line match regex",
        "second line match regex"
      ],

**Reply by otbe** [reply]
As a side note: 
regex and example code was taken from the documentation -> https://docs.renovatebot.com/modules/datasource/#aws-machine-image-datasource

**Comment by Gabriel-Ladzaretti** [comment, marked as answer]
It looks like you are missing a region.
after adding `{region: 'us-west-2'}`
```
# amiFilter=[{"Name":"owner-id","Values":["762873164855"]},{"Name":"name","Values":["ops-base-master-*"]},{region: 'us-west-2'}]
# currentImageName=unknown
base_image_id = "ami-04a726a285fbf6c8d"
```
i was able to query the datasource:
```
DEBUG: packageFiles with updates (repository=ladzaretti/AMI)
       "config": {
         "regex": [
           {
             "packageFile": "noxfile.py",
             "deps": [
               {
                 "depName": "base_image_id",
                 "packageName": "[{\"Name\":\"owner-id\",\"Values\":[\"762873164855\"]},{\"Name\":\"name\",\"Values\":[\"ops-base-master-*\"]},{region: 'us-west-2'}]\r",
                 "currentValue": "ami-04a726a285fbf6c8d",
                 "currentDigest": "unknown\r",
                 "datasource": "aws-machine-image",
                 "replaceString": "# amiFilter=[{\"Name\":\"owner-id\",\"Values\":[\"762873164855\"]},{\"Name\":\"name\",\"Values\":[\"ops-base-master-*\"]},{region: 'us-west-2'}]\r\n# currentImageName=unknown\r\nbase_image
_id = \"ami-04a726a285fbf6c8d\"\r",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "base_image_id",
                     "message": "Failed to look up dependency base_image_id"
                   }
                 ],
                 "versioning": "aws-machine-image"
               }
             ],
             "matchStrings": [
               ".*amiFilter=(?<packageName>.*?)\n(.*currentImageName=(?<currentDigest>.*?)\n)?(.*\n)?.*?(?<depName>[a-zA-Z0-9-_:]*)[ ]*?[:|=][ ]*?[\"|']?(?<currentValue>ami-[a-z0-9]{17})[\"|']?.*"
             ],
             "datasourceTemplate": "aws-machine-image"
           }
         ]
       }

```

**Reply by asierga95** [reply]
Thanks! This helped us. Also we were testing with the onboarding branch created by Renovate in Gitlab, where is possible to change the renovate.json file but apart from this file, the rest of the files are checked against master.

**Reply by Gabriel-Ladzaretti** [reply]
If i understand correctly, you can set the base branches via [baseBranches](https://docs.renovatebot.com/configuration-options/#basebranches). 
e.g:
```json
"baseBranches": ["main", "dev"]
```
